dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -319,8 +319,13 @@ protected void githubWebhookDelete(String repository, String gitReference, Strin
         });
 
         // Delete all non-frozen versions that have the same git reference name and then update the file formats of the entry.
+        List<String> entryNamesWithDeletedVersions = new ArrayList<>();
         workflows.forEach(workflow -> {
-            workflow.getWorkflowVersions().removeIf(workflowVersion -> Objects.equals(workflowVersion.getName(), gitReferenceName.get()) && !workflowVersion.isFrozen());
+            Predicate<WorkflowVersion> canVersionBeDeleted = workflowVersion -> Objects.equals(workflowVersion.getName(), gitReferenceName.get()) && !workflowVersion.isFrozen();
```

Comment:
Is this the change from 
> While writing the tests, I noticed that the endpoint that installs the GitHub app, handleGitHubInstallation, throws an| exception if the GitHub releases of the few branches it finds is unsuccessful. 

Or did you encounter multiple versions with the same name?

Reply:
Oh, my previous comments were lost when the PR was updated :thinking: I'll add the comment again.

This change is unrelated to that issue. This was done so that I could get a list of workflow names that were affected by the deletion. The `Predicate` was created so that it's less likely for the predicate to change in one place and not the other (i.e. we updated the predicate in the `removeIf` line, but forgot to do it in the `if (workflow.getWorkflowVersions().stream().anyMatch(<predicate>))` line